### PR TITLE
Update poller: add support for scalerURL

### PR DESCRIPTION
**Pull request by afarbos**
The current behavior hard code the scaler url to `http://scaler`.
This PR add support for custom scaler URL while keeping the same behavior by default.

Upgrade tests from nodejs 10 to 18 since `||=` is not supported by the current version.
This is an actively supported version and used in the Dockerfile of the repository

### CODE REVIEW

**Review by benpoliquin — APPROVED** [review]
will use this feature, lgtm 😄

**Review by ross-bryan — APPROVED** [review]
lgtm

**Comment by spidercensus** [comment]
Great feature, we will certainly use it.  I wonder if there is a sleeker way to define this once so that it doesn't need to be re-defined for every spanner instance in the config.

File: poller/poller-core/index.js [hunk 1]
```diff
@@ -217,9 +217,12 @@ function postPubSubMessage(spanner, metrics) {
 }
 
 function callScalerHTTP(spanner, metrics) {
+  spanner.scalerURL ||= 'http://scaler';
+  const url = new URL('/metrics', spanner.scalerURL);
+
```
Comment (henrybell) [inline poller/poller-core/index.js:222]: Nit: possible to remove the whitespace on this line?

Comment (afarbos) [inline poller/poller-core/index.js:222]:
```suggestion
```

Comment (afarbos) [inline poller/poller-core/index.js:222]:
```suggestion

```

Comment (afarbos) [reply poller/poller-core/index.js]: done!

**Comment by afarbos** [comment]
@henrybell I did run the test locally and it seems to work fine. With the error here I am wondering if this is a flaky test? or am I missing something?

**Comment by henrybell** [comment]
> @henrybell I did run the test locally and it seems to work fine. With the error here I am wondering if this is a flaky test? or am I missing something?

Likewise, works locally for me as well. Will look into it.

**Comment by henrybell** [comment]
> > @henrybell I did run the test locally and it seems to work fine. With the error here I am wondering if this is a flaky test? or am I missing something?
> 
> Likewise, works locally for me as well. Will look into it.

Looks like the GItHub Action that runs the tests is using an older version of NodeJS that predates support for Logical OR assignment `(x ||= y)`, which was introduced in version 15. I'll pursue updating this, and once done we should be good to go.

**Comment by afarbos** [comment]
> > > @henrybell I did run the test locally and it seems to work fine. With the error here I am wondering if this is a flaky test? or am I missing something?
> > 
> > 
> > Likewise, works locally for me as well. Will look into it.
> 
> Looks like the GitHub Action that runs the tests is using an older version of NodeJS that predates support for Logical OR assignment `(x ||= y)`, which was introduced in version 15. I'll pursue updating this, and once done we should be good to go.

I see, I updated the version here if that helps.

**Comment by henrybell** [comment]
There are some more changes required -- I've discussed with my colleagues and should be able to update early next week.

**Comment by afarbos** [comment]
> There are some more changes required -- I've discussed with my colleagues and should be able to update early next week.

@henrybell I saw you merged #124, I updated this branch. It's now ready for approval again.

**Comment by henrybell** [comment]
Thanks!

**Pull request merged**